beacon-chain/state/stategen/migrate.go

```diff
@@ -37,76 +38,87 @@ func (s *State) MigrateToCold(ctx context.Context, fRoot [32]byte) error {
 		return nil
 	}
 
-	// Start at previous finalized slot, stop at current finalized slot (it will be handled in the next migration).
-	// If the slot is on archived point, save the state of that slot to the DB.
-	for slot := oldFSlot; slot < fSlot; slot++ {
+	// Calculate the first archived point slot >= oldFSlot (but > 0).
+	// This avoids iterating through every slot and only visits archived points directly.
+	var startSlot primitives.Slot
+	if oldFSlot == 0 {
+		startSlot = s.slotsPerArchivedPoint
+	} else if oldFSlot%s.slotsPerArchivedPoint == 0 {
+		// oldFSlot is already on an archived point, start from oldFSlot itself
+		startSlot = oldFSlot
+	} else {
+		// Round up to the next archived point
+		startSlot = oldFSlot + (s.slotsPerArchivedPoint - oldFSlot%s.slotsPerArchivedPoint)
+	}
+
```

Comment:
This can all be simplified to 

```suggestion
	var startSlot primitives.Slot
	if oldFSlot == 0 {
		startSlot = s.slotsPerArchivedPoint
	} else 
		// Round up to the next archived point
		startSlot = oldFSlot + s.slotsPerArchivedPoint - oldFSlot%s.slotsPerArchivedPoint
	}
```

Reply:
When `oldFSlot` is already on an archived boundary (e.g. 2048) then it will miss the first one. 
So instead of having [2048, 4096, 6144, 8192] it will output [4096, 6144, 8192]

Reply:
Oh sorry  I left your computation in the second branch. It should be 


```suggestion
	var startSlot primitives.Slot
	if oldFSlot == 0 {
		startSlot = s.slotsPerArchivedPoint
	} else {
		// Round up to the next archived point
		startSlot = (oldFSlot + s.slotsPerArchivedPoint - 1) / s.slotsPerArchivedPoint * s.slotsPerArchivedPoint
	}

```